clippy_dev/src/update_lints.rs

```diff
@@ -220,26 +230,62 @@ pub fn rename(old_name: &str, new_name: &str, uplift: bool) {
             .then_with(|| lhs.old_name.cmp(&rhs.old_name))
     });
 
-    if found_new_name {
+    if uplift {
         println!(
-            "`{}` is already defined. Removal the old lint code in `clippy_lints` will have to be done manually",
+            "`{}` has be uplifted. Please remove all the code inside `clippy_lints` related to it.",
+            old_name
+        );
+        let contents = gen_renamed_lints_test(&renamed_lints);
+        fs::write("tests/ui/rename.rs", contents.as_bytes())
+            .unwrap_or_else(|e| panic!("error writing file `tests/ui/rename.rs`: {}", e));
+    } else if found_new_name {
+        println!(
+            "`{}` is already defined. The old linting code inside `clippy_lints` has to be updated/removed manually",
             new_name
         );
         let contents = gen_renamed_lints_test(&renamed_lints);
         fs::write("tests/ui/rename.rs", contents.as_bytes())
             .unwrap_or_else(|e| panic!("error writing file `tests/ui/rename.rs`: {}", e));
     } else {
-        // Update the lint struct name
-        lints[old_lint_index].name = new_name.into();
-        let old_name = old_name.to_uppercase();
-        let new_name = new_name.to_uppercase();
+        // Rename the lint struct and the containing module if it shares the lint name across all files.
+        let lint = &mut lints[old_lint_index];
+        let old_name_upper = old_name.to_uppercase();
+        let new_name_upper = new_name.to_uppercase();
+        let new_module_path = PathBuf::from(format!("clippy_lints/src/{}.rs", new_name));
+        let replacements;
+
+        let (replacements, mut renamed_file) = if lint.module == lint.name
+            // Only rename the module if the file doesn't already exist.
+            && let Ok(file) = fs::OpenOptions::new().create_new(true).write(true).open(&new_module_path)
+        {
+            lint.module = new_name.into();
+            replacements = [(&*old_name_upper, &*new_name_upper), (old_name, new_name)];
+            (replacements.as_slice(), Some(file))
+        } else {
+            replacements = [(&*old_name_upper, &*new_name_upper), ("", "")];
+            (&replacements[0..1], None)
+        };
+        lint.name = new_name.into();
 
-        for (_, file) in clippy_lints_src_files() {
+        for (rel_path, file) in clippy_lints_src_files() {
             let path = file.path();
             let contents =
                 fs::read_to_string(path).unwrap_or_else(|e| panic!("Cannot read from `{}`: {}", path.display(), e));
-            if contents.contains(&old_name) {
-                fs::write(path, contents.replace(&old_name, &new_name).as_bytes())
+            let new_contents = replace_ident_like(&contents, replacements);
+
+            if let Some(ref mut file) = renamed_file
+                // Make sure the file is directly in the src directory and not a sub-directory.
+                && rel_path.parent().map_or(false, |p| p.as_os_str().is_empty())
```

Comment:
It's actually blocked earlier by the check making sure the module name is the same as the lint name. This is more of a defensive check in case something weird happens.